nav2_amcl/src/amcl_node.cpp

```diff
@@ -845,18 +845,20 @@ AmclNode::laserReceived(sensor_msgs::msg::LaserScan::ConstSharedPtr laser_scan)
       // p.covariance[6*5+5] = hyps[max_weight_hyp].pf_pose_cov.m[2][2];
       p.pose.covariance[6 * 5 + 5] = set->cov.m[2][2];
 
-      /*
-         printf("cov:\n");
-         for(int i=0; i<6; i++)
-         {
-         for(int j=0; j<6; j++)
-         printf("%6.3f ", p.covariance[6*i+j]);
-         puts("");
-         }
-       */
+      float temp = 0.0;
+      for (int i = 0; i < static_cast<int>(p.pose.covariance.size()); i++) {
+        temp += p.pose.covariance[i];
```

Comment:
Clever way of doing that, I like it. Out of curiousity, why not also `+=` `p.pose.pose.position.x` and `p.pose.pose.position.x` to reduce the nan checks to 1?

Reply:
That's right. That would make less line of codes.  Will update.